django/db/models/fields/related.py

```diff
@@ -340,8 +341,13 @@ def get_reverse_related_filter(self, obj):
             rh_field.attname: getattr(obj, lh_field.attname)
             for lh_field, rh_field in self.related_fields
         }
-        base_filter.update(self.get_extra_descriptor_filter(obj) or {})
-        return base_filter
+        descriptor_filter = self.get_extra_descriptor_filter(obj)
+        base_q = Q(**base_filter)
+        if isinstance(descriptor_filter, dict):
+            return base_q & Q(**descriptor_filter)
+        elif descriptor_filter:
+            return base_q & descriptor_filter
```

Comment:
Is this branch tested? No tests seem to fail (I tired SQLite & PG) if it's removed.